src/components/containeredInput/containeredInput.ts

```diff
@@ -0,0 +1,110 @@
+import {blockProperty} from "../block/types";
+import {Block} from "../block/block";
+import template from "./template";
+import {Label} from "../label/label";
+import {Input} from "../Input/input";
+import {Templator} from "../../utils/Templator/Templator";
+
+import "./style.scss";
+
+
+export class ContaineredInput extends Block {
+  
+  fieldNameLabel: Label;
+  
+  inputField: Input;
+  
+  validationLabel: Label;
+  
+  alwaysShowlabel: boolean;
+  
+  neverShowlabel: boolean;
+  
+  constructor(props: blockProperty) {
+    const fieldNameLabel = new Label({classes: props.labelClasses || [], textContent: props.placeholder || "Поле"});
+    if (!props.alwaysShowlabel) {
+      fieldNameLabel.hide();
+    }
+    const inputField = new Input(props);
+    const validationLabel = new Label({classes: ["base-label", "warning-label", ...props.warningLabelClasses || []], textContent: ""});
+    validationLabel.hide();
+    
+    super("div", {
+      classes: [
+        "input-container",
+        ...(props.containerClasses || [])
+      ],
+      fieldNameLabel,
+      inputField,
+      validationLabel
+    }, template);
+    this.alwaysShowlabel = props.alwaysShowlabel;
+    this.neverShowlabel = props.neverShowlabel;
+    this.fieldNameLabel = fieldNameLabel;
+    const newListeners = Object.assign(props.listeners || {},
+      {
+        focus: (): void => {
+          inputField.focused = true;
+          this.checkHideLabel();
+          this.validate();
+        },
+        blur: (e: Event): void => {
+          inputField.setProps({value: (e.target as HTMLInputElement).value});
+          inputField.focused = false;
+          this.checkHideLabel();
+          this.validate();
+        }
+      });
```

Comment:
Можно использовать деструктуризацию
```suggestion
    const newListeners =  {
        ...props.listeners,
        focus: (): void => {
          inputField.focused = true;
          this.checkHideLabel();
          this.validate();
        },
        blur: (e: Event): void => {
          inputField.setProps({value: (e.target as HTMLInputElement).value});
          inputField.focused = false;
          this.checkHideLabel();
          this.validate();
    };
```